ui/packages/consul-ui/tests/integration/adapters/topology-test.js

```diff
@@ -22,16 +23,18 @@ module('Integration | Adapter | topology', function(hooks) {
   test("requestForQueryRecord throws if you don't specify an id", function(assert) {
     const adapter = this.owner.lookup('adapter:topology');
     const client = this.owner.lookup('service:client/http');
+    const request = client.url.bind(client);
     assert.throws(function() {
-      adapter.requestForQueryRecord(client.url, {
+      adapter.requestForQueryRecord(request, {
         dc: dc,
       });
     });
   });
   test('requestForQueryRecord returns the correct body', function(assert) {
     return nspaceRunner(
       (adapter, serializer, client) => {
-        return adapter.requestForQueryRecord(client.body, {
+        const request = client.body.bind(client);
+        return adapter.requestForQueryRecord(request, {
```

Comment:
So, all this test editing was a bit tedious 😄 , got me thinking about how we could improve this somehow.